### PR TITLE
more mypy checks, docopt-->argparse, formatting, cleanup, ci improvements, ...

### PR DESCRIPTION
**Pull request by goerlibe**

### CODE REVIEW

File: discopop_explorer/pattern_detectors/task_parallelism/tp_utils.py [hunk 1]
```diff
@@ -431,9 +431,7 @@ def __get_word_prior_to_bracket(string):
         string = string[-1]
         return string
 
-    called_function_name_contained = False
-    if called_function_name is None:
-        called_function_name_contained = True
+    called_function_name_contained = called_function_name is None
```
Comment (goerlibe) [inline discopop_explorer/pattern_detectors/task_parallelism/tp_utils.py:434]: i think we can simplify further and get rid of the variable entirely, but it is late and my brain is not 100% sure

File: discopop_explorer/PETGraphX.py [hunk 2]
```diff
@@ -1154,7 +1152,7 @@ def unused_check_alias(self, s: NodeID, t: NodeID, d: Dependency, root_loop: Nod
         parent_func_source = self.get_parent_function(self.node_at(t))
 
         res = False
-        d_var_name_str = cast(str, str(d.var_name))
+        d_var_name_str = str(d.var_name)
```
Comment (goerlibe) [inline discopop_explorer/PETGraphX.py:1155]: this seems weird... 
`str(None)` becomes "None" string in python, and I do not think this is what we want....?

**Comment by goerlibe** [comment]
I think the settings need some change to accomodate for the removed "CI Tests Successful" job. However I do not have sufficient access rights for that.

**Comment by goerlibe** [comment]
run ---> __run

File: discopop_explorer/PETGraphX.py (hunk 2 above)
Comment (goerlibe) [reply discopop_explorer/PETGraphX.py]: fixed

**Comment by goerlibe** [comment]
unless you see any issues, I think this is ready for merging. @lukasrothenberger

**Review by lukasrothenberger — APPROVED** [review]

File: discopop_library/discopop_optimizer/scheduling/workload_delta.py [hunk 3]
```diff
@@ -224,15 +224,7 @@ def __parse_node(
...
-        min_wl = (
-            0
-            if cast(Workload, node_data).sequential_workload is None
-            else cast(int, cast(Workload, node_data).sequential_workload)
-        ) + (
-            0
-            if cast(Workload, node_data).parallelizable_workload is None
-            else cast(int, cast(Workload, node_data).parallelizable_workload)
-        )
+        min_wl = (node_data.sequential_workload or 0) + (node_data.parallelizable_workload or 0)
```
Comment (lukasrothenberger) [inline discopop_library/discopop_optimizer/scheduling/workload_delta.py:227]: nice solution! :)

**Review by lukasrothenberger — APPROVED** [review]

**Pull request merged**